keyboards/crkbd/keymaps/colemak_luna/keymap.c

```diff
@@ -0,0 +1,405 @@
+    render_logo();
+if (current_wpm > 0) {
+    anim_sleep = timer_read32();
+    /* wpm counter */
+    uint8_t n = get_current_wpm();
+    char    wpm_str[4];
+    oled_set_cursor(0, 14);
+    wpm_str[3] = '\0';
+    wpm_str[2] = '0' + n % 10;
+    wpm_str[1] = '0' + (n /= 10) % 10;
+    wpm_str[0] = '0' + n / 10;
+    oled_write(wpm_str, false);
+
+    oled_set_cursor(0, 15);
+    oled_write(" wpm", false);
+
+ /* this fixes the screen on and off bug */
+   
+    } else if(timer_elapsed32(anim_sleep) > OLED_TIMEOUT) {
+        /* clear */
+        oled_set_cursor(0,0);
+        oled_write("                                                                                                                        ", false);
+        oled_off();
+       
+   
+    }
+}
+
+static void print_status_narrow(void) {
+   
+
+    /* Print current layer */
+    oled_write("LAYER", false);
+
+    oled_set_cursor(0, 6);
+
+    switch (get_highest_layer(layer_state)) {
+        case 0:
+            oled_write("Base ", false);
+            rgb_matrix_reload_from_eeprom();
+            break;
+        case 1:
+            oled_write("Game ", false);
+            rgb_matrix_mode_noeeprom(RGB_MATRIX_ALPHAS_MODS);
+                rgb_matrix_sethsv_noeeprom(HSV_TEAL);
+            break;
+        case 2:
+            oled_write("Game2", false);
+            rgb_matrix_mode_noeeprom(RGB_MATRIX_ALPHAS_MODS);
+            rgb_matrix_sethsv_noeeprom(HSV_PURPLE);
+            break;
+        case 3:
+            oled_write("Lower", false);
+            //rgb_matrix_mode_noeeprom(RGB_MATRIX_ALPHAS_MODS);
+            //rgb_matrix_sethsv_noeeprom(HSV_TEAL);
+            break;
+        case 4:
+            oled_write("Raise", false);
+            //rgb_matrix_mode_noeeprom(RGB_MATRIX_ALPHAS_MODS);
+            //rgb_matrix_sethsv_noeeprom(HSV_RED);
+            break;
+        case 5:
+            oled_write("Adj  ", false);
+            rgb_matrix_mode_noeeprom(RGB_MATRIX_ALPHAS_MODS);
+            rgb_matrix_sethsv_noeeprom(HSV_YELLOW);
+            break;
+        default:
+            oled_write("Undef", false);
+    }
+
+
+    /* KEYBOARD PET RENDER START */
+
+    render_luna(0, 13);
+
+    /* KEYBOARD PET RENDER END */
+}
+
+oled_rotation_t oled_init_user(oled_rotation_t rotation) { return OLED_ROTATION_270; }
+
+bool oled_task_user(void) {
+    /* KEYBOARD PET VARIABLES START */
+
+    current_wpm   = get_current_wpm();
+    led_usb_state = host_keyboard_led_state();
+
+    /* KEYBOARD PET VARIABLES END */
+
+    if (is_keyboard_master()) {
+        print_status_narrow();
+    } else {
+        print_logo_narrow();
+    }
+    return false;
+}
+
+#endif
+ bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    switch (keycode) {
+        /* KEYBOARD PET STATUS START */
+ 
+        case KC_LCTL:
+            if (record->event.pressed) {
+                isSneaking = true;
+            } else {
+                isSneaking = false;
+            }
```

Comment:
All you need is this:
```suggestion
            isSneaking = record->event.pressed;
```

Also, you can always use `get_mods() & MOD_MASK_CTRL` wherever you need to see if control is active, instead.